File: GENERECIDAD /EJERCICIO1/JAVA/Caja.java
class Caja<T> {
    private T contenido;

    public void guardar(T valor) {
        this.contenido = valor;
    }

    public T obtener() {
        return contenido;
    }
}

public class Main {
    public static void main(String[] args) {
        // Caja para un String
        Caja<String> cajaDeTexto = new Caja<>();
        cajaDeTexto.guardar("Astro Aroha");

        // Caja para un Integer
        Caja<Integer> cajaDeNumero = new Caja<>();
        cajaDeNumero.guardar(123);

        // Mostrar contenido
        System.out.println("Contenido de cajaDeTexto: " + cajaDeTexto.obtener());
        System.out.println("Contenido de cajaDeNumero: " + cajaDeNumero.obtener());
    }
}

File: GENERECIDAD /EJERCICIO1/PYTHON/Caja.py
from typing import Generic, TypeVar

T = TypeVar('T')

class Caja(Generic[T]):
    def __init__(self):
        self.contenido: T = None

    def guardar(self, valor: T):
        self.contenido = valor

    def obtener(self) -> T:
        return self.contenido

# Caja para un String
caja_de_texto = Caja[str]()
caja_de_texto.guardar("Astro Aroha")

# Caja para un Integer
caja_de_numero = Caja[int]()
caja_de_numero.guardar(123)

# Mostrar contenido
print("Contenido de caja_de_texto:", caja_de_texto.obtener())
print("Contenido de caja_de_numero:", caja_de_numero.obtener())
